src/permissions/permissions.service.ts

```diff
@@ -0,0 +1,128 @@
+/*
+ * SPDX-FileCopyrightText: 2021 The HedgeDoc developers (see AUTHORS file)
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+
+import { Injectable } from '@nestjs/common';
+import { User } from '../users/user.entity';
+import { Note } from '../notes/note.entity';
+
+export enum GuestPermission {
+  DENY = 'deny',
+  READ = 'read',
+  WRITE = 'write',
+  CREATE = 'create',
+  CREATE_ALIAS = 'createAlias',
+}
+
+@Injectable()
+export class PermissionsService {
+  public guestPermission: GuestPermission; // TODO cange to configOption
+  mayRead(user: User, note: Note): boolean {
+    if (this.isOwner(user, note)) return true;
+
+    if (this.hasPermissionUser(user, note, false)) return true;
+
+    // noinspection RedundantIfStatementJS
+    if (this.hasPermissionGroup(user, note, false)) return true;
+
+    return false;
+  }
+
+  mayWrite(user: User, note: Note): boolean {
+    if (this.isOwner(user, note)) return true;
+
+    if (this.hasPermissionUser(user, note, true)) return true;
+
+    // noinspection RedundantIfStatementJS
+    if (this.hasPermissionGroup(user, note, true)) return true;
+
+    return false;
+  }
+
+  mayCreate(user: User): boolean {
+    if (user) {
+      return true;
+    } else {
+      if (
+        this.guestPermission == GuestPermission.CREATE ||
+        this.guestPermission == GuestPermission.CREATE_ALIAS
+      ) {
+        // TODO change to guestPermission to config option
```

Comment:
Should this still be here?

Reply:
Yes because config option is not defined by now.